zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/mutable/MutableAuthorizationState.java

```diff
@@ -9,8 +9,29 @@
 
 import io.camunda.zeebe.engine.state.immutable.AuthorizationState;
 import io.camunda.zeebe.protocol.impl.record.value.authorization.AuthorizationRecord;
+import io.camunda.zeebe.protocol.record.value.AuthorizationResourceType;
+import io.camunda.zeebe.protocol.record.value.PermissionType;
+import java.util.List;
 
 public interface MutableAuthorizationState extends AuthorizationState {
 
   void createAuthorization(final AuthorizationRecord authorizationRecord);
+
+  /**
+   * Checks if an Authorization exists for the provided ownerKey, resourceType and permissionType.
+   * If it does, adds the permissions to this entry. If it does not, creates a new entry with the
+   * provided permissions.
+   *
+   * @param ownerKey the key of the owner of the permissions. This could be a userKey, a roleKey or
+   *     a groupKey
+   * @param resourceType the type of resource the permissions are for (Eg. Process definition, Job)
+   * @param permissionType The type of permission being granted (Eg. READ, WRITE)
+   * @param resourceIds A list of resourceIds the permissions are granted for (Eg.
+   *     bpmnProcessId:foo, *)
+   */
+  void createOrAddPermissions(
```

Comment:
❓ Does this method add/create single permission to multiple resources ? We can change the name in `createOrAddPermission` and adjust the javadoc accordingly